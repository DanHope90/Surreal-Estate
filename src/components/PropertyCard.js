import React from "react";
import PropTypes from "prop-types";
import "../styles/PropertyCard.css";

function PropertyCard(props) {
  const { title, city, type, bedrooms, bathrooms, price, email } = props;
  return (
    <div className="property-card">
      <div className="title" data-testid="card-title">
        {title}
      </div>
      <div className="city" data-testid="card-city">
        {city}
      </div>
      <div className="type" data-testid="card-type">
        {type}
      </div>
      <div className="bedrooms" data-testid="card-bedrooms">
        Bedrooms:{bedrooms}
      </div>
      <div className="bathrooms" data-testid="card-bathrooms">
        Bathrooms:{bathrooms}
      </div>
      <div className="price" data-testid="card-price">
        Â£:{price}
      </div>
      <div className="email" data-testid="cardId-email">
        <a href={`mailto:${email}`}>Send Email</a>
      </div>
    </div>
  );
}

export default PropertyCard;

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  bedrooms: PropTypes.number.isRequired,
  bathrooms: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  email: PropTypes.string.isRequired,
};
