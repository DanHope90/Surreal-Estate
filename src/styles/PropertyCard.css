.property-card {
font-weight: bold;
  background-color: white;
  margin: 20px;
  border: 2px solid black;
  display: flex;
  flex-direction: column;
}

.email {
    padding: 5px;
    text-align: center;
    background-color: rgb(47, 162, 190);
}