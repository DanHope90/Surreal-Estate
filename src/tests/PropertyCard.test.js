import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    fields: {
      title: "Stub Title",
      city: "Stub City",
      type: "Stub Type",
      bedrooms: 2,
      bathrooms: 1,
      price: 100000,
      email: "mock@email.com",
    },
  };

  it("renders the correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={validProps.fields.title}
        city={validProps.fields.city}
        type={validProps.fields.type}
        bedrooms={validProps.fields.bedrooms}
        bathrooms={validProps.fields.bathrooms}
        price={validProps.fields.price}
        email={validProps.fields.email}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders correct prop values", () => {
    const { getByText, getByTestId } = render(
      <PropertyCard
        title={validProps.fields.title}
        city={validProps.fields.city}
        type={validProps.fields.type}
        bedrooms={validProps.fields.bedrooms}
        bathrooms={validProps.fields.bathrooms}
        price={validProps.fields.price}
        email={validProps.fields.email}
      />
    );
    expect(getByText("Stub Title")).toHaveClass("title");
    expect(getByText("Stub City")).toHaveClass("city");
    expect(getByText("Stub Type")).toHaveClass("type");
    expect(getByText(1)).toHaveClass("bathrooms");
    expect(getByText(2)).toHaveClass("bedrooms");
    expect(getByText(100000)).toHaveClass("price");
    expect(getByTestId("cardId-email")).toHaveClass("email");
  });
});
