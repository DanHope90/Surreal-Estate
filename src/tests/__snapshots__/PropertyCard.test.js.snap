// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders the correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="title"
      data-testid="card-title"
    >
      Stub Title
    </div>
    <div
      class="city"
      data-testid="card-city"
    >
      Stub City
    </div>
    <div
      class="type"
      data-testid="card-type"
    >
      Stub Type
    </div>
    <div
      class="bedrooms"
      data-testid="card-bedrooms"
    >
      Bedrooms:2
    </div>
    <div
      class="bathrooms"
      data-testid="card-bathrooms"
    >
      Bathrooms:1
    </div>
    <div
      class="price"
      data-testid="card-price"
    >
      Â£:100000
    </div>
    <div
      class="email"
      data-testid="cardId-email"
    >
      <a
        href="mailto:mock@email.com"
      >
        Send Email
      </a>
    </div>
  </div>
</DocumentFragment>
`;
